### PR TITLE
add parameter f

### LINKED ISSUE #3: Parameter -F

### PR DESCRIPTION
**Pull request by justinmueller05082000**
fixes #3

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -26,6 +27,7 @@ func (i *arrayFlags) Set(value string) error {
 func main() {
 	var flagTest arrayFlags
 	flag.Var(&flagTest, "e", "use PATTERN for matching.")
+	fixedStringsParameter := flag.Bool("F", false, "PATTERN is a set of newline-separated strings.")
```
Comment (Al2Klimov) [inline main.go]: Don't just copy&paste from the manual. Describe what **your** program does.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
`go run . -F -e '*'` doesn't work.

File: main.go [hunk 2]
```diff
@@ -57,13 +60,25 @@ func main() {
...
-			if i.Match(data) {
-				if !bytes.HasSuffix(data, []byte{'\n'}) {
-					data = append(data, '\n')
+		if *fixedStringsParameter {
+			for _, i := range flagTest {
+				if strings.Contains(string(data), i) {
+					if !bytes.HasSuffix(data, []byte{'\n'}) {
+						data = append(data, '\n')
+					}
+					_, _ = os.Stdout.Write(data)
```
Comment (Al2Klimov) [inline main.go]: De-duplicate this block.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

File: main.go [hunk 3]
```diff
@@ -57,13 +60,23 @@ func main() {
 			break
 		}
 
-		for _, i := range regArr {
-			if i.Match(data) {
-				if !bytes.HasSuffix(data, []byte{'\n'}) {
-					data = append(data, '\n')
+		if !bytes.HasSuffix(data, []byte{'\n'}) {
+			data = append(data, '\n')
+		}
```
Comment (Al2Klimov) [inline main.go]: You've separated this block from its container if which I didn't ask for, ...

File: main.go [hunk 4]
```diff
@@ -57,13 +60,23 @@ func main() {
...
+
+		if *fixedStringsParameter {
+			for _, i := range flagTest {
+				if strings.Contains(string(data), i) {
+					_, _ = os.Stdout.Write(data)
```
Comment (Al2Klimov) [inline main.go]: ... but didn't completely de-duplicate the code block I asked to de-duplicate.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

File: main.go [hunk 5]
```diff
@@ -57,16 +61,28 @@ func main() {
...
 		}
 
-		for _, i := range regArr {
-			if i.Match(data) {
-				if !bytes.HasSuffix(data, []byte{'\n'}) {
-					data = append(data, '\n')
+		if *fixedStringsParameter {
+			for _, i := range flagTest {
+				if strings.Contains(string(data), i) {
+					equalComparator = true
```
Comment (Al2Klimov) [inline main.go:67]: Once you know that it matches one -e, why you continue checking the other -e's?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
* [x] `git reset --soft HEAD~$(git rev-list --count $(git merge-base HEAD master)..HEAD^)`
* [x] `git commit --amend --no-edit`
* [x] `git push -f origin feature/parameter-f-3`

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Write a test protocol. Prove that both single and multiple -e (2) and both -F and no -F (2) work (2x2=4).

**Comment by justinmueller05082000** [comment]
**Test protocol:**

**toGrep.txt:**

```
fo*bar dog cat cat foobar dog cat mouse mouse
tiger dog cat dog mouse mouse zebra elephant
dog cat foobar mouse ze*bra zebra cat dog
tiger dog cat mouse dog dog cat zebra zebra tiger
elephant foobar foobar elephant foobar
```

**Case (1):**

**Terminal:** `go run . -F -e 'fo*bar' -e 'ze*bra' < toGrep.txt`

**Output:**

```
fo*bar dog cat cat foobar dog cat mouse mouse
dog cat foobar mouse ze*bra zebra cat dog
```

**Grep command:**

**Terminal:**  `grep -F -e 'fo*bar' -e 'ze*bra' < toGrep.txt`

**Output:**

```
fo*bar dog cat cat foobar dog cat mouse mouse
dog cat foobar mouse ze*bra zebra cat dog
```

____________________________________

**Case (2):**

**Terminal:** `go run . -e 'fo*bar' -e 'ze*bra' < toGrep.txt`

**Output:**

```
fo*bar dog cat cat foobar dog cat mouse mouse
tiger dog cat dog mouse mouse zebra elephant
dog cat foobar mouse ze*bra zebra cat dog
tiger dog cat mouse dog dog cat zebra zebra tiger
elephant foobar foobar elephant foobar
```

**Grep command:**

**Terminal:**  `grep -e 'fo*bar' -e 'ze*bra' < toGrep.txt`

**Output:**

```
fo*bar dog cat cat foobar dog cat mouse mouse
tiger dog cat dog mouse mouse zebra elephant
dog cat foobar mouse ze*bra zebra cat dog
tiger dog cat mouse dog dog cat zebra zebra tiger
elephant foobar foobar elephant foobar
```

____________________________________

**Case (3):**

**Terminal:** `go run . -F -e 'fo*bar' < toGrep.txt`

**Output:**

```
fo*bar dog cat cat foobar dog cat mouse mouse
```

**Grep command:**

**Terminal:**  `grep -F -e 'fo*bar' < toGrep.txt`

**Output:**

```
fo*bar dog cat cat foobar dog cat mouse mouse
```
____________________________________

**Case (4):**

**Terminal:** `go run . -e 'fo*bar' < toGrep.txt`

**Output:**

```
fo*bar dog cat cat foobar dog cat mouse mouse
dog cat foobar mouse ze*bra zebra cat dog
elephant foobar foobar elephant foobar
```

**Grep command:**

**Terminal:**  `grep -e 'fo*bar' < toGrep.txt`

**Output:**

```
fo*bar dog cat cat foobar dog cat mouse mouse
dog cat foobar mouse ze*bra zebra cat dog
elephant foobar foobar elephant foobar
```
____________________________________

**Conclusion:**
All results are equal except for the colored highlighting in the terminal.

**Review by Al2Klimov — APPROVED** [review]

**Pull request merged**